Tests/Tests/STPRedirectContextTest.m

```diff
@@ -282,4 +282,58 @@ - (void)testSafariAppRedirectFlow_noNotification {
     OCMReject([sut dismissPresentedViewController]);
 }
 
+/**
+ If a source type that supports native redirect is used and it contains a native
+ url, an app to app redirect should attempt to be initiated.
+ */
+- (void)testNativeRedirectSupportingSourceFlow_validNativeURL {
+    STPSource *source = [STPFixtures alipaySourceWithNativeUrl];
+
+    STPRedirectContext *context = [[STPRedirectContext alloc] initWithSource:source
+                                                                  completion:^(__unused NSString *sourceID, __unused NSString *clientSecret, __unused NSError *error) {
+        XCTFail(@"completion called");
+    }];
+    id sut = OCMPartialMock(context);
+
+    id applicationMock = OCMClassMock([UIApplication class]);
+    OCMStub([applicationMock sharedApplication]).andReturn(applicationMock);
+
+    id mockVC = OCMClassMock([UIViewController class]);
+    [context startRedirectFlowFromViewController:mockVC];
+
+    OCMReject([sut startSafariViewControllerRedirectFlowFromViewController:[OCMArg any]]);
+    OCMReject([sut startSafariAppRedirectFlow]);
+    OCMVerify([applicationMock openURL:[OCMArg any]
+                               options:[OCMArg any]
+                     completionHandler:[OCMArg any]]);
```

Comment:
In the past, I've been able to use an `[OCMArg checkWithBlock:]` to trigger the completion to test the entire flow:

```
OCMVerify([applicationMock openURL:options:completionHandler:[OCMArg checkWithBlock:^BOOL(void (^)completion) {
  completion();
}]]);
```